libraries/aws_kms_key.rb

```diff
@@ -71,4 +74,19 @@ def has_rotation_enabled?
   def to_s
     opts.key?(:aws_region) ? "KMS Key #{@display_name} in #{opts[:aws_region]}" : "KMS Key #{@display_name}"
   end
+
+  def fetch_key_id
+    catch_aws_errors do
+      key_id = ''
+      response = @aws.kms_client.list_aliases
+      return [] if !response || response.empty?
+      response.aliases.each do |alias_entry|
+        if alias_entry['alias_name'] == @alias
+          key_id = alias_entry['target_key_id'] if alias_entry['target_key_id']
```

Comment:
You should return directly from here, otherwise you will continue to iterate after assignment.
```
return alias_entry['target_key_id'] if alias_entry['target_key_id']
```